plugins/zenoh-plugin-storage-manager/src/replica/digest.rs

```diff
@@ -492,9 +494,12 @@ impl Digest {
                     .unwrap()
                     .content
                     .retain(|x| {
-                        x.timestamp.get_time() != entry.timestamp.get_time()
-                            && x.timestamp.get_id() != entry.timestamp.get_id()
-                            && x.key != entry.key
+                        let timestamp_identical =
+                            x.timestamp.get_time() == entry.timestamp.get_time() &&
+                            x.timestamp.get_id() == entry.timestamp.get_id();
+                        let key_identical = x.key == entry.key;
+                        let element_identical = timestamp_identical && key_identical;
+                        !element_identical
```

Comment:
This could be simplified like so:
```suggestion
                        x.timestamp != entry.timestamp || x.key != entry.key
```

Reply:
Done